brayns/utils/image/Image.cpp

```diff
@@ -81,7 +82,22 @@ size_t Image::getPixelSize() const
 
 void Image::write(const Image &image, size_t x, size_t y)
 {
-    write(image.getData(), image.getSize(), x, y);
+    if (image.getChannelCount() != getChannelCount() || image.getChannelSize() != getChannelSize())
+    {
+        throw std::invalid_argument("Images have different shape");
+    }
+
+    if (image.getWidth() + x > getWidth() || image.getHeight() + y > getHeight())
+    {
+        throw std::out_of_range("Image write overflow");
+    }
+
```

Comment:
For me this is an invalid usage so I would just put an assert and not test that but it is ok like that also...

Reply:
For me its an API, so it should be handled at runtime. The assert part is very specific to the usage we make of it in Brayns, but since it is not a performance critical part of the code (we do not even use it), I prefer to go with the first part which is a more correct design.